Testing/Heroes/GargoyleTests.cs

```diff
@@ -0,0 +1,241 @@
+    {
+        #region Gargoyle Utilities
+        private HeroTurnTakerController gargoyle => FindHero("Gargoyle");
+        private Card mobileDefensePlatform => GetCardInPlay("MobileDefensePlatform");
+        private string[] gameDecks => new string[] { "BaronBlade", "Cauldron.Gargoyle", "Unity", "Bunker", "TheScholar", "Megalopolis" };
+
+        private void StartTestGame()
+        {
+            SetupGameController(gameDecks);
+            StartGame();
+
+            DestroyCard(mobileDefensePlatform);
+        }
+
+        private void SetupIncapTest()
+        {
+            StartTestGame();
+            SetupIncap(baron);
+            AssertIncapacitated(gargoyle);
+            GoToUseIncapacitatedAbilityPhase(gargoyle);
+        }
+
+        private void SetupIncap(TurnTakerController villain)
+        {
+            SetHitPoints(gargoyle.CharacterCard, 1);
+            DealDamage(villain, gargoyle, 2, DamageType.Melee);
+        }
+
+        #endregion Gargoyle Utilities
+
+        [Test()]
+        public void TestLoadGargoyle()
+        {
+            SetupGameController(gameDecks);
+
+            Assert.AreEqual(6, this.GameController.TurnTakerControllers.Count());
+
+            Assert.IsNotNull(gargoyle);
+            Assert.IsInstanceOf(typeof(GargoyleCharacterCardController), gargoyle.CharacterCardController);
+
+            Assert.AreEqual(27, gargoyle.CharacterCard.HitPoints);
+        }
+
+        /*
+         * Power:
+         * "Select a target. Reduce the next damage it deals by 1. Increase the next damage Gargoyle deals by 1."
+         */
+        [Test()]
+        public void TestGargoyleInnatePower()
+        {
+            StartTestGame();
+
+            GoToUsePowerPhase(gargoyle);
+
+            // Select a target. 
+            DecisionSelectTarget = baron.CharacterCard;
+            UsePower(gargoyle.CharacterCard);
+                        
+            // Reduce the next damage it deals by 1.
+            QuickHPStorage(gargoyle.CharacterCard);
+            DealDamage(baron.CharacterCard, gargoyle.CharacterCard, 1, DamageType.Melee);
+            QuickHPCheckZero();
+
+            // Increase the next damage Gargoyle deals by 1.
+            QuickHPStorage(baron.CharacterCard);
+            DealDamage(gargoyle.CharacterCard, baron.CharacterCard, 1, DamageType.Melee);
+            QuickHPCheck(-2);
+        }
+
+        /* 
+         * Incap 1
+         * "Increase the next damage dealt by a hero target by 2."
+        */
+        [Test()]
+        public void TestGargoyleIncap1HeroCharacter()
+        {
+            SetupIncapTest();
+            UseIncapacitatedAbility(gargoyle, 0);
+
+            // Make sure it only affects hero damage
+            QuickHPStorage(baron);
+            DealDamage(bunker.CharacterCard, baron, 1, DamageType.Melee);
+            QuickHPCheck(-3);
+
+            // Should only be the next damage,  this packet of damage should not be increased
+            QuickHPStorage(baron);
```

Comment:
good practice is to stick all the hero's and villains into the array and ensure that there are no damage side effects.  Not a big deal on these simple cases, but has come up.